### ISSUE TITLE
`pad_packed_sequence` is not exactly the inverse of `pack_padded_sequence`

### ISSUE THREAD

**Issue by dozed** [pytorch/pytorch#65890]
## 🐛 Bug

When a padded sequence is received from a batch it is possible that the last timesteps of all sequences are padded with zero. In that case packing und unpacking the sequences will drop the last timestep. Basically the packing/unpacking looses information.

## To Reproduce

```python
import torch
from torch.nn.utils.rnn import pack_padded_sequence, pad_packed_sequence

padded_batch = torch.LongTensor(
    [[1, 2, 3, 0],
     [5, 6, 0, 0],
     [7, 0, 0, 0]]
)

seqs_len = [3, 2, 1]

packed_batch = pack_padded_sequence(padded_batch, seqs_len, batch_first=True)
print(packed_batch)
# PackedSequence(data=tensor([1, 5, 7, 2, 6, 3]), batch_sizes=tensor([3, 2, 1]), sorted_indices=None, unsorted_indices=None)

padded_batch1, _ = pad_packed_sequence(packed_batch, batch_first=True, padding_value=0)
print(padded_batch1)
# note that the last paddings are dropped
# tensor([[1, 2, 3],
#         [5, 6, 0],
#         [7, 0, 0]])
```

This behaviour requires further handling in downstream components.
For example when doing sequence labeling it is required to adapt the label sequence before computing the loss:

```python
padded_batch_seq_len = padded_batch.shape[1]
padded_batch1_seq_len = padded_batch1.shape[1]

if padded_batch_seq_len != padded_batch1_seq_len:
    pass
    # shorten label sequence here
```

## Expected behavior

I would have expected the unpacked tensor to be the same as the original one.

## Environment

 - PyTorch Version (e.g., 1.9.1):
 - OS (e.g., Linux): Ubuntu 20.04.3 LTS
 - How you installed PyTorch (`conda`, `pip`, source): conda
 - Python version: 3.8.11
 - CUDA/cuDNN version: 10.2.89/7.6.5
